integration/tests/basic/test_contract_reverting.py

```diff
@@ -60,31 +69,68 @@ def test_constructor_raises_no_argument_error(self, solc_version):
         )
 
         with pytest.raises(
-            web3.exceptions.ContractLogicError, match="execution reverted"
+                web3.exceptions.ContractLogicError, match="execution reverted"
         ):
             contract.constructor([]).build_transaction()
 
-    def test_method_raises_string_based_error(self):
-        contract, _ = self.web3_client.deploy_and_get_contract(
-            "common/TrivialRevert",
-            "0.7.0",
-            self.sender_account,
-            constructor_args=[],
-        )
+    def test_method_raises_string_based_error(self, revert_contract):
         with pytest.raises(
-            web3.exceptions.ContractLogicError,
-            match="execution reverted: Predefined revert happened",
+                web3.exceptions.ContractLogicError,
+                match="execution reverted: Predefined revert happened",
         ):
-            contract.functions.do_string_based_revert().call()
+            revert_contract.functions.doStringBasedRevert().call()
 
-    def test_method_raises_trivial_error(self):
+    def test_method_raises_trivial_error(self, revert_contract):
+        with pytest.raises(
+                web3.exceptions.ContractLogicError, match="execution reverted"
+        ):
+            revert_contract.functions.doTrivialRevert().call()
+
+    def test_nested_contract_revert(self, revert_contract):
         contract, _ = self.web3_client.deploy_and_get_contract(
-            "common/TrivialRevert",
-            "0.7.0",
-            self.sender_account,
-            constructor_args=[],
+            contract="common/Revert", version="0.8.10",
+            contract_name="Caller", account=self.sender_account,
+            constructor_args=[revert_contract.address]
         )
         with pytest.raises(
-            web3.exceptions.ContractLogicError, match="execution reverted"
+                web3.exceptions.ContractLogicError,
+                match="execution reverted: Predefined revert happened"
+        ):
+            contract.functions.doStringBasedRevert().call()
+
+    def test_eth_call_revert(self, revert_contract):
+        tx = {
+            "from": self.sender_account.address,
+            "to": revert_contract.address,
+            "data": revert_contract.encodeABI(fn_name="doStringBasedRevert")
```

Comment:
you don't need this tx, because web3 provide API to call:
```
revert_contract.functions. doStringBasedRevert().call()
```

Check documentation: https://web3py.readthedocs.io/en/stable/web3.contract.html by work: eth_call

Reply:
In `test_method_raises_string_based_error` test we have such api call. Does this mean I can remove `test_eth_call_revert`?

Reply:
No, string method should use build_transaction, this test case should use .call()

Reply:
Done